fmlloader/src/main/java/net/minecraftforge/fml/loading/RuntimeDistCleaner.java

```diff
@@ -138,21 +144,71 @@ public int processClassWithFlags(final Phase phase, final ClassNode classNode, f
 
     @SuppressWarnings("unchecked")
     private static List<AnnotationNode> unpack(final List<AnnotationNode> anns) {
-        if (anns == null) return Collections.emptyList();
-        List<AnnotationNode> ret = anns.stream().filter(ann->Objects.equals(ann.desc, ONLYIN)).collect(Collectors.toList());
-        anns.stream().filter(ann->Objects.equals(ann.desc, ONLYINS) && ann.values != null)
-            .map( ann -> (List<AnnotationNode>)ann.values.get(ann.values.indexOf("value") + 1))
-            .filter(v -> v != null)
-            .forEach(v -> v.forEach(ret::add));
-        return ret;
+        if (anns == null) {
+            return Collections.emptyList();
+        }
+
+        List<AnnotationNode> unpacked = new ArrayList<>();
+
+        for (var annotationNode : anns) {
+            if (Objects.equals(annotationNode.desc, ONLYINS)) {
+                unpacked.add(annotationNode);
+
+                if (annotationNode.values != null) {
+                    List<AnnotationNode> subNodes = (List<AnnotationNode>) annotationNode.values.get(annotationNode.values.indexOf("value") + 1);
+
+                    if (subNodes != null) {
+                        unpacked.addAll(subNodes);
+                    }
+                }
+            }
+        }
+
+        return unpacked;
     }
 
     private boolean remove(final List<AnnotationNode> anns, final String side)
```

Comment:
Braces on same line for this method please

Reply:
Done